File: src/Lession4/Ex4.java
package Lession4;

import java.util.Scanner;

public class Ex4 {

    static private final int MAXLENGTH = 10;

    static void main() {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        for (int k = 0; k < n; k++) {
            String str = sc.next();
            System.out.println(reduce(str));
        }
    }

    static String reduce (String inputString) {
        if (inputString.length() <= MAXLENGTH) {
            return inputString;
        }

        StringBuilder sc = new StringBuilder();
        sc.append(inputString.charAt(0));
        sc.append(inputString.length()-2);
        sc.append(inputString.charAt(inputString.length() - 1));
        return sc.toString();
    }
}

File: src/Lession4/Ex4.kt
package Lession4

const val MAXLENGTH = 10

fun main() {
    val n = readln().toInt()

    for (i in 0 until n) {
        val str = readln()
        println(reduce(str))
    }
}

fun reduce (inputString: String): String {
    if (inputString.length <= MAXLENGTH) {
        return inputString
    }

    val sb = StringBuilder()
    sb.append(inputString[0])
    sb.append(inputString.length - 2)
    sb.append(inputString[inputString.length - 1])

    return sb.toString()
}
